### PR TITLE
Receive bitfield on new peer connection

### PR DESCRIPTION
**Pull request by matei-oltean**

### CODE REVIEW

File: messaging/messages.go [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+	cancel
+)
+
+// Message represents a message: its type and payload
+type Message struct {
+	Type    messageType
+	Payload []byte
+}
+
+// Read reads a message coming from a connection
```
Comment (Nicolas-Ding) [inline messaging/messages.go]: is that a typo ?

File: messaging/messages.go [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+func Read(reader io.Reader) (*Message, error) {
+	// A message is composed as follows:
```
Comment (Nicolas-Ding) [inline messaging/messages.go:34]: as follow*

Comment (matei-oltean) [reply messaging/messages.go:34]: It is always *follows*, see https://dictionary.cambridge.org/dictionary/english/as-follows

File: messaging/messages.go (hunk 1 above)
Comment (matei-oltean) [reply messaging/messages.go]: No :|

**Pull request merged**